cyw43/src/runner.rs

```diff
@@ -73,6 +73,41 @@ where
         }
     }
 
+    async fn validate_firmware(&mut self, ram_addr: u32, firmware: &[u8]) {
+        debug!("validating firmware");
+        let mut i = 0;
+        let mut mem_bytes = [0; 0x100];
+        while i < firmware.len() {
+            let slice_len = if 0x100 <= firmware.len() - i {
+                0x100
+            } else {
+                firmware.len() - i
+            };
+            self.bus
+                .bp_read((ram_addr as usize + i) as u32, &mut mem_bytes[0..slice_len])
+                .await;
+            let firmware_slice = &firmware[i..i + slice_len];
+            let mem_slice = &mem_bytes[..slice_len];
+            assert_eq!(firmware_slice.len(), mem_slice.len());
+            for j in 0..firmware_slice.len() {
+                if firmware_slice[j] != mem_slice[j] {
+                    debug!(
+                        "{:08x} firmware_slice[{}] != mem_slice[{}] {:02x} != {:02x}",
+                        ram_addr as usize + i + j,
+                        j,
+                        j,
+                        firmware_slice[j],
+                        mem_slice[j]
+                    );
+                }
+                assert_eq!(firmware_slice[j], mem_slice[j]);
```

Comment:
@Dirbaio would you expect this to fail under any conditions / what would you say to me if it was?

It sounds like you are saying it doesn't matter if firmware is `probe-rs` written in or not because no matter what at boot it's going to get `self.bus.bp_write(ram_addr, firmware).await;` written to `0x00000000`?